extensions/amp-story/1.0/page-advancement.js

```diff
@@ -289,32 +304,78 @@ class ManualAdvancement extends AdvancementConfig {
   }
 
   /** @override */
-  start() {
-    super.start();
-    this.element_.addEventListener('click', this.clickListener_, true);
+  getProgress() {
+    return 1.0;
   }
 
-  /** @override */
-  stop() {
-    super.stop();
-    this.element_.removeEventListener('click', this.clickListener_, true);
+  /**
+   * Binds the event listeners.
+   * @private
+   */
+  startListening_() {
+    if (isExperimentOn(this.win_, 'amp-story-hold-to-pause')) {
+      this.element_
+          .addEventListener('touchstart', this.onTouchstart_.bind(this), true);
+      this.element_
+          .addEventListener('touchend', this.onTouchend_.bind(this), true);
+    }
+
+    this.element_
+        .addEventListener(
+            'click', this.maybePerformNavigation_.bind(this), true);
   }
 
-  /** @override */
-  getProgress() {
-    return 1.0;
+  /**
+   * TouchEvent touchstart events handler.
+   * @param {!Event} event
+   * @private
+   */
+  onTouchstart_(event) {
+    // Don't start the paused state if the target is not a descendant of an
+    // amp-story-page. Also ignores any subsequent touchstart that would happen
+    // before touchend was fired, since it'd reset the touchstartTimestamp (ie:
+    // user touches the screen with a second finger).
+    if (this.touchstartTimestamp_ || !this.isAmpStoryPageDescendant_(event)) {
+      return;
+    }
+
+    this.touchstartTimestamp_ = Date.now();
+    this.storeService_.dispatch(Action.TOGGLE_PAUSED, true);
+  }
+
+  /**
+   * TouchEvent touchend events handler.
+   * @param {!Event} event
+   * @private
+   */
+  onTouchend_(event) {
+    // Ignores the event if there's still a user's finger holding the screen.
+    const touchesCount = (event.touches || []).length;
+    if (!this.touchstartTimestamp_ || touchesCount > 0) {
+      return;
+    }
+
+    // Cancels the navigation if user paused the story for over 500ms. Calling
+    // preventDefault on the touchend event ensures the click/tap event won't
+    // fire.
+    if ((Date.now() - this.touchstartTimestamp_) > 500) {
```

Comment:
Extract to constant.

Anecdotally, I feel like this should probably be shorter than 500ms.